PixelHunter1995/GameStates/IGameState.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace PixelHunter1995.GameStates
+{
+    /// <summary>
+    /// All kind of states the game can be in.
+    /// </summary>
+    interface IGameState
+    {
+        IGameState Update(GameTime gameTime, Dictionary<String, IGameState> gameStates);
```

Comment:
As I think you mentioned, it would be nicer with some kind of StateManager instead of a dictionary. Could you do that? And instead of having a dictionary, the StateManager could have specific functions for different states (I don't think we will have so many that it will be unreasonable). That way, we can know at compile time if we're trying to change into a state that does not exist.

Reply:
If we use a StateManager, then it would be reasonable that it knows of the current state. Then, we would not have to return a state from this function.